src/sbvr-api/permissions.coffee

```diff
@@ -124,7 +125,31 @@ exports.setup = (app, sbvrUtils) ->
 		POSTPARSE: ({ req, request }) ->
 			# If the abstract sql query is already generated then adding permissions will do nothing
 			return if request.abstractSqlQuery?
+			if (request.method == 'POST' and request.odataQuery.property?.resource == 'canAccess')
+				request.odataQuery.options = { '$select': { 'properties': [ { 'name': 'id' } ] }, $top: 1 }
+				{ action, method } = request.values
+				if method? and action?
+					throw new MissingOrInvalidArgumentsError()
+				else if method? and methodPermissions[method]?
+					request.permissionType = methodPermissions[method]
+				else if action?
+					request.permissionType = action
+				else
+					throw new MissingOrInvalidArgumentsError()
+				request.resourceName = request.resourceName.slice(0, request.resourceName.length - '#canAccess'.length)
+				request.odataQuery.resource = request.resourceName
+				delete request.odataQuery.property
+				request.method = 'GET'
+				request.custom.isAction = 'canAccess'
 			addPermissions(req, request)
+		PRERESPOND: ({ request, data }) ->
+			if (request.custom.isAction == 'canAccess')
+				if (_.isArray(data))
+					for i in [0..data.length - 1]
+						data[i] = _.pick(data[i], 'id', '__metadata')
```

Comment:
What fields are you excluding here btw? As the `$select` you add should already cover this I thought?

Reply:
I'm concerned, that with an `expand` linked resources could be fetched. Although I was not able to build a query for this, so that it fetches sub resources. But since I'm not certain that's why I added this filter.